### PR TITLE
Add support for x86_64-fortanix-unknown-sgx target

### PR DESCRIPTION
**Pull request by akash-fortanix**

### CODE REVIEW

File: src/sys.rs [hunk 1]
```diff
@@ -270,6 +271,85 @@ mod inner {
...
+    pub fn utc_tm_to_time(tm: &Tm) -> i64 {
+        tm_to_time(tm)
+    }
+
+    pub fn local_tm_to_time(tm: &Tm) -> i64 {
+        // FIXME: Add timezone logic
+        tm_to_time(tm)
+    }
+
+    pub fn get_time() -> (i64, i32) {
```
Comment (jethrogb) [inline src/sys.rs:308]: Write this in terms of `SteadyTime::new`

File: src/sys.rs [hunk 2]
```diff
@@ -270,6 +271,85 @@ mod inner {
...
+        match SystemTime::now().duration_since(SystemTime::UNIX_EPOCH) {
+            Ok(n) => (n.as_secs() as i64, n.subsec_nanos() as i32),
+            Err(_) => panic!("SystemTime before UNIX EPOCH!"),
+        }
+    }
+
+    pub fn get_precise_ns() -> u64 {
+        match SystemTime::now().duration_since(SystemTime::UNIX_EPOCH) {
+            Ok(n) => n.as_nanos() as u64,
```
Comment (jethrogb) [inline src/sys.rs]: This function is not stable, compute manually instead from `SystemTime::now()`

File: src/sys.rs [hunk 3]
```diff
@@ -270,6 +271,85 @@ mod inner {
...
+    }
+
+    impl SteadyTime {
+        pub fn now() -> SteadyTime {
+            match SystemTime::now().duration_since(SystemTime::UNIX_EPOCH) {
+                Ok(n) => match Duration::from_std(n) {
+                    Ok(d) => SteadyTime { t: d },
+                    Err(_) => panic!("original duration is larger than the maximum value supported"),
+                },
+                Err(_) => panic!("SystemTime before UNIX EPOCH!"),
```
Comment (jethrogb) [inline src/sys.rs]: Use `unwrap` for error handling, add comments explaining why `unwrap` is ok

File: src/sys.rs [hunk 4]
```diff
@@ -270,6 +271,85 @@ mod inner {
...
+            }
+        }
+    }
+
+    impl Sub for SteadyTime {
+        type Output = Duration;
+        fn sub(self, _other: SteadyTime) -> Duration {
```
Comment (jethrogb) [inline src/sys.rs]: Remove `_` prefix

**Review by jethrogb — CHANGES_REQUESTED** [review]

File: src/duration.rs [hunk 5]
```diff
@@ -285,6 +285,11 @@ impl Duration {
         }
         Ok(StdDuration::new(self.secs as u64, self.nanos as u32))
     }
+
+    /// Returns the raw value of duration.
+    pub fn raw(&self) -> (i64, i32) {
```
Comment (jethrogb) [inline src/duration.rs]: Should be `pub(crate)`

File: src/sys.rs [hunk 6]
```diff
@@ -270,6 +271,81 @@ mod inner {
...
+        tm_to_time(tm)
+    }
+
+    pub fn get_time() -> (i64, i32) {
+        SteadyTime::now().t.raw()
+    }
+
+    pub fn get_precise_ns() -> u64 {
+        let duration = SteadyTime::now().t.raw();
+        duration.0 as u64 * NANOS_PER_SEC + duration.1 as u64
```
Comment (jethrogb) [inline src/sys.rs]: This is fine. My original suggestion was to do this one in terms of `SystemTime`, not `SteadyTime`, that would save you a signed-unsigned conversion.

**Comment by akash-fortanix** [comment]
filed PR against original repo, the purpose of this PR is served

**Pull request closed without merging**